code/parachain/frame/composable-maths/src/dex/constant_product.rs

```diff
@@ -129,7 +129,9 @@ where
 /// To get `a_sent` without accounting for the fee, set `f = 0`.
 /// Amount in, round up results.
 ///
-/// **NOTE:** Weights must already be normalized.
+/// Notes:
+/// * Weights must already be normalized
+/// * `1 - f` must be less than `2^96`
```

Comment:
We could add some debug assertions to check these

Reply:
The function correctly errors in this case. However, this is not what I meant to document here. Will update